test/e2e/dataplane/service_discovery.go

```diff
@@ -42,4 +50,69 @@ func RunServiceDiscoveryTest(f *framework.Framework) {
 
 	Expect(err).NotTo(HaveOccurred())
 	Expect(stdout).To(ContainSubstring("Welcome to nginx!"))
+
+	kubeDnsService, _ := f.ClusterClients[framework.ClusterB].CoreV1().Services("kube-system").Get("kube-dns", metav1.GetOptions{})
+	kubeDnsServiceIP := kubeDnsService.Spec.ClusterIP
+
+	cmd = []string{"dig", "@" + kubeDnsServiceIP, "nginx-demo." + f.Namespace + ".svc.cluster2.local", "+short"}
```

Comment:
```suggestion
	cmd = []string{"dig", "@" + kubeDnsServiceIP, nginxServiceClusterC.Name + "." + f.Namespace + ".svc.cluster2.local", "+short"}
```

I think it would be useful to add a `By`, ie

`By(fmt.Sprintf("Executing %q to verify discovery of Nginx service IP %q", strings.Join(cmd, " "), nginxServiceClusterCIP))`

Reply:
Done.